externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/SparkOperation.scala

```diff
@@ -193,6 +208,25 @@ abstract class SparkOperation(session: Session)
     // TODO: (fchen) make all operation support arrow
     getClass.getCanonicalName == classOf[ExecuteStatement].getCanonicalName
   }
+
+  protected def setSessionUserSign(): Unit = {
+    (
+      session.conf.get(KYUUBI_SESSION_SIGN_PUBLICKEY),
+      session.conf.get(KYUUBI_SESSION_USER_SIGN)) match {
+      case (Some(pubKey), Some(userSign)) =>
+        setSparkLocalProperty(KYUUBI_SESSION_SIGN_PUBLICKEY, pubKey)
```

Comment:
>  Server -> Engine: Server generates session signing on session user when opening Kyuubi Session to Engine

Since each session user has its own `KYUUBI_SESSION_USER_SIGN`, but all the user sessions within one application share one `SparkContext`, this means the verification mechanism only works for CONNECTION/USER mode?

Reply:
It should work for SERVER share level as well.
1. setting properties to SparkContext is to put value in a ThreadLocal prop holders, not sharing with other threads.
2. `runInternal` is running  the code in synchronized, which means  `setSessionUserSign` and `claeaSessionUserSign` properly set/clear the properties in the same thread  right before/after the code excuted